internal/pkg/cli/pipeline_init.go

```diff
@@ -355,27 +353,56 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) selectGitHubURL() error {
+	url, err := opts.prompt.SelectOne(
+		pipelineSelectGitHubURLPrompt,
+		pipelineSelectGitHubURLHelpPrompt,
+		opts.repoURLs,
 	)
-
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+	if err != nil {
+		return fmt.Errorf("select GitHub URL: %w", err)
 	}
+	opts.GitHubURL = url
 
-	if err != nil {
-		return fmt.Errorf("failed to get GitHub repository: %w", err)
+	return nil
+}
+
+func (opts *InitPipelineOpts) parseOwnerRepoName(url string) (string, string, error) {
+	parsedURL := strings.TrimPrefix(url, opts.regexZeroLengthAssertion(url, `.*(github.com)(:|\/)`, "", ""))
+	ownerRepo := strings.Split(parsedURL, string(os.PathSeparator))
+	if len(ownerRepo) != 2 {
+		return "", "", fmt.Errorf(`unable to parse the owner and repository for "%s", please pass in your repository URL with -u`, url)
 	}
+	return ownerRepo[0], ownerRepo[1], nil
+}
 
-	opts.GitHubRepo = repo
+func (opts *InitPipelineOpts) regexZeroLengthAssertion(s string, pattern string, prefix string, suffix string) string {
+	regexPattern := regexp.MustCompile(pattern)
+	regexResult := regexPattern.FindString(s)
+	resultWithoutSuffixAndPrefix := strings.TrimSuffix(strings.TrimPrefix(regexResult, prefix), suffix)
+	return resultWithoutSuffixAndPrefix
+}
 
-	return nil
+// examples:
+// efekarakus	git@github.com:efekarakus/grit.git (fetch)
+// efekarakus	https://github.com/karakuse/grit.git (fetch)
+// origin	https://github.com/koke/grit (fetch)
+// koke      git://github.com/koke/grit.git (push)
+func (opts *InitPipelineOpts) parseGitRemoteResult(s string) ([]string, error) {
+	var urls []string
+	urlSet := make(map[string]bool)
+	items := strings.Split(s, "\n")
+	for _, item := range items {
+		url := strings.TrimSpace(opts.regexZeroLengthAssertion(item, `\s(http|git).* `, "\r", ".git "))
```

Comment:
Do we need this pattern? could we not get rid of this logic now?

Reply:
Do you mean get rid of `regexZeroLengthAssertion` this function and move the logic out? Or just don't use regex?

Reply:
Why can't we just do:
```go
for _, row := range rows {
    cols := strings.Split(row, "\t")
    url := strings.TrimSpace(strings.TrimSuffix(cols[1], ".git"))
}
```

Reply:
Yeah you are right, previously I thought we should only have those urls start with https or git, but apparently we should not help them to exam if they have a valid url. We should instead just get what we want.

Reply:
However, the reason why I want to use regex is because if we use regex and any invalid input we can just don't match them. And empty match string means it is invalid. Using the script that you mentioned might result in goroutine panic when there is an invalid input. I am not sure if this is a good reason for using regex.